Dockerfile

```
@@ -65,4 +67,45 @@ RUN mkdir /root/sgx-rust && \
 # TODO: remove once all the code supports the later nightly toolchains
 RUN /root/.cargo/bin/rustup toolchain install nightly-2021-03-25
 
+FROM teaclave-base AS builder
+ENV SGX_SDK=/opt/sgxsdk
 WORKDIR /root
+
+COPY --from=teaclave-base /root/sgx-rust/ ./sgx-rust/
+COPY --from=teaclave-base /root/.cargo/ ./.cargo/
+
+WORKDIR /root/rtc-data
+ARG SGX_MODE=HW
+
+COPY . .
+RUN --mount=type=cache,target=/root/.cargo/registry \
+    --mount=type=cache,target=/root/.cargo/git \
+    --mount=type=cache,sharing=private,target=/root/rtc-data/target \
+    --mount=type=cache,sharing=private,target=/root/rtc-data/rtc_data_enclave/target \
+    ./runbuild.sh
+
+RUN --mount=type=cache,sharing=private,target=/root/rtc-data/target \
```

```suggestion
    ./runbuild.sh && \
```

The `target` cache mounts can be deleted or change arbitrarily between two RUN commands, so copying the `http_server` executable out should happen within the same mount lifetime.